### PR TITLE
Feature/push/changes to integrate with test app

### PR DESCRIPTION
**Pull request by marto83**
Fairly large commit which integrates the mobile app with Ably push notifications

### CODE REVIEW

**Review by tomkirbygreen — CHANGES_REQUESTED** [review]
I think we should start writing doc-comments (`///` etc) for classes and at least `public` APIs now. We need to have such at the end, and starting to add it now can only aid others in understanding the code and it helps reduce any operational risk for the business by capturing more of your understanding in the code.

File: examples/DotnetPush/DotnetPush/App.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        protected override void OnStart()
+        {
+        }
+
+        protected override void OnSleep()
+        {
+        }
+
+        protected override void OnResume()
+        {
```
[the remark this answers is not in the document]
Comment (marto83) [reply examples/DotnetPush/DotnetPush/App.xaml.cs:44]: That came with the template. I'm going to remove them.

File: examples/DotnetPush/DotnetPush.Android/MyFirebaseMessaging.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
+﻿using Android.App;
+using Android.Content;
+using Android.Support.V4.App;
+using Firebase.Messaging;
+using System;
```
[the remark this answers is not in the document]
Comment (marto83) [reply examples/DotnetPush/DotnetPush.Android/MyFirebaseMessaging.cs:5]: Oops. I'm going to add style cop to the project so it picks up on stuff like that.

[the remark this answers is not in the document]
Comment (marto83) [reply examples/DotnetPush/DotnetPush.Android/MyFirebaseMessaging.cs:5]: The rule is namespaces should be in alphabetical order. If we change it in the Stylecop file we will need to update it everywhere :)

**Pull request closed without merging**